application/common/models/User.php

```diff
@@ -292,7 +292,7 @@ private function validatePassword(): Closure
     {
         return function ($attributeName) {
             $currentPassword = $this->currentPassword ?? new Password();
-            if (!password_verify($this->password, $currentPassword->hash)) {
+            if ($currentPassword->hash === null || !password_verify($this->password, $currentPassword->hash)) {
```

Comment:
Are you sure you can access hash on new Password()? I would have ?? null, and then ?->hash.

Reply:
No, you'd know better than I. However, this has been working for years until now that PHP is complaining about the second parameter of `password_verify` not accepting null.

Reply:
I'm just worried new Password() may not initialize hash and then your check might fail. This is why I suggested the change to 294 (?? null) and then the new check (?->hash).

Reply:
If you can tell me the line of code where a `new Password()` initializes the `hash` property to `null`, then as is is okay. Otherwise, I'd recommend the coding change I suggested. PHP 8.x is finicky about accessing variables before they are initialized.

Reply:
Is this adequate? 
```php
    public function rules(): array
    {
        return ArrayHelper::merge([
// ...
            [
                'hash', 'default', 'value' => function () {
                    return self::hashPassword($this->password);
                },
            ],
// ...

        ], parent::rules());
    }
```

https://github.com/silinternational/idp-id-broker/blob/3ac20ea34735c529e5bed123769467e6ac875d10/application/common/models/Password.php#L56

Reply:
The problem is rules are generally only triggered on validation. There's no validation between new Password() and the logic below it. Models have their init() called on construction.

Reply:
OK, thanks. I tried it in PHP interactive shell, and the `?->` didn't seem to help. So I just added `?? null`.